whotracksme/website/builder.py

```diff
@@ -112,62 +114,67 @@ def feed_event(self, event):
-                futures.append(executor.submit(build_privacy_policy, data=data_source))
+                build_home(data=data_source)
+                build_privacy_policy(data=data_source)
 
                 # Trackers
-                futures.append(executor.submit(build_trackers_list, data=data_source))
-                futures.append(executor.submit(build_tracker_pages, data=data_source))
+                trackers = [id for id, _ in data_source.trackers.iter()]
+                batched_job(trackers, build_tracker_page_batch, 50, "Generate tracker pages")
 
                 # Websites
-                futures.append(executor.submit(build_website_list, data=data_source))
-                futures.append(executor.submit(build_website_pages, data=data_source))
+                websites = list(enumerate([id for id, _ in data_source.sites.iter()]))
+                batched_job(websites, build_website_pages_batch, 100, "Generate website pages")
+                build_website_list(data=data_source)
 
                 # Companies
-                futures.append(executor.submit(build_company_reach_chart_page, data=data_source))
+                build_company_reach_chart_page(data=data_source)
 
             # Depends on: 'data/', 'blog/', 'templates/'
             if event & DATA_FOLDER or event & BLOG_FOLDER or event & TEMPLATES_FOLDER:
                 futures.append(executor.submit(
-                    build_blogpost_list,
-                    data=data_source,
+                    build_blogpost_pages,
                     blog_posts=self.blog_posts
                 ))
 
-                futures.append(executor.submit(
-                    build_blogpost_pages,
+                build_blogpost_list(
                     data=data_source,
                     blog_posts=self.blog_posts
-                ))
+                )
 
             # Depends on: 'data/', 'blog/', 'templates/'
             if event & DATA_FOLDER or event & BLOG_FOLDER or event & TEMPLATES_FOLDER:
                 futures.append(executor.submit(
                     generate_sitemap,
-                    data=data_source,
                     blog_posts=self.blog_posts
                 ))
 
-            if event & DATA_FOLDER:
-                futures.append(executor.submit(
-                    build_tracker_db
-                ))
-                futures.append(executor.submit(
-                    build_api,
-                    data=data_source,
-                ))
-
             # Explorer: depends on 'data/'
             if event & DATA_FOLDER or event & STATIC_FOLDER:
                 futures.append(executor.submit(
                     build_explorer,
-                    data=data_source,
                 ))
 
+            if event & DATA_FOLDER:
+                futures.append(executor.submit(
+                    build_tracker_db
+                ))
+                trackers = [id for id, _ in data_source.trackers.iter()]
```

Comment:
Should this run after the `build_tracker_db` function is done?

Reply:
No, `build_tracker_db` is just for the JSON database published on the site, not the database used by the build process.